types/geoNamesSchema.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from "zod";
+
+export const geoNamesItem = z.object({
+  geonameId: z.number(),
+  name: z.string(),
+  countryName: z.string(),
+  countryCode: z.string(),
+  lat: z.number(),
+  lng: z.number(),
+});
+
+export const dataSchema = z.object({
+  geonames: z.array(geoNamesItem),
+});
+
+export type GeoNamesData = z.infer<typeof dataSchema>;
```

Comment:
Ta schema jest używana tylko do inferowania typu. To jest ten sam problem, który mieliśmy wcześniej przy pogodzie - w ten sposób `zod` nam nic nie daje. I tak castujemy ten typ. Tak się można wkopać w problemy. 
```TS
const dataFromUnknownSource: any = { a: 2, b: "3" }
const dataSchema = z.object({ a: z.number(), b: z.number() })
type Data = z.infer<typeof dataSchema>

const data: Data = dataFromUnknownSource

// ups! TS uważa, że data.b to number, podczas gdy data.b to string
```

Reply:
dodalem parsowanie w locationSearch